### ISSUE TITLE
Crash when changing the settings

### ISSUE THREAD

**Issue by joyful-cambion** [Hydr8gon/NooDS#49]
I have found a bug in this, it occurs when changing any setting option while in mid game, after you get back to your game and click on touch screen it always crashes the emulator
-

**Comment by Hydr8gon** [comment]
What platform is this on? Android?

**Comment by joyful-cambion** [comment]
yes Android, it happened when changing "screen filter" from on to off also it sometimes happen with other settings too like turning off "fps counter" and changing "fps limiter"

**Comment by FrostKnight** [comment]
> ## I have found a bug in this, it occurs when changing any setting option while in mid game, after you get back to your game and click on touch screen it always crashes the emulator

Funny you should say this...

The last release does the same on Hyperbola and it could do it on other distros as well...

Not sure, but its worth checking..

;)

**Comment by Hydr8gon** [comment]
Yeah, there are some threading issues that can cause crashes on the desktop UI. I've been meaning to look into these, but it's hard to track them all down when they're not consistent.
